sdk/internal/recording/recording.go

```diff
@@ -438,7 +439,37 @@ var modeMap = map[RecordMode]recorder.Mode{
 	Playback: recorder.ModeReplaying,
 }
 
-var recordMode = os.Getenv("AZURE_RECORD_MODE")
+func init() {
+	recordMode = os.Getenv("AZURE_RECORD_MODE")
+	if !(recordMode == RecordingMode || recordMode == PlaybackMode || recordMode == LiveMode) {
+		log.Printf("AZURE_RECORD_MODE was not understood, options are %s, %s, or %s Received: %v. Defaulting to playback", RecordingMode, PlaybackMode, LiveMode, recordMode)
```

Comment:
I think we should set a default only when AZURE_RECORD_MODE has no value. When it has an unexpected one we know the user wants specific behavior but not what behavior. Erroring out so the user can clarify seems better to me than proceeding.